Client/Frontend/Library/Bookmarks/BookmarksPanel.swift

```diff
@@ -355,16 +356,25 @@ class BookmarksPanel: SiteTableViewController,
                 !(node is BookmarkSeparatorData),
                 isCurrentFolderEditable(at: indexPath) {
                 // Only show detail controller for editable nodes
-                showBookmarkDetailController(for: node, folder: bookmarkFolder)
+                if CoordinatorFlagManager.isLibraryCoordinatorEnabled {
+                    bookmarkCoordinatorDelegate?.showBookmarkDetail(for: node, folder: bookmarkFolder)
+                    updatePanelState(newState: .bookmarks(state: .itemEditMode))
+                } else {
+                    showBookmarkDetailController(for: node, folder: bookmarkFolder)
+                }
             }
             return
         }
 
         updatePanelState(newState: .bookmarks(state: .inFolder))
-        bookmarkCell.didSelect(profile: profile,
-                               libraryPanelDelegate: libraryPanelDelegate,
-                               navigationController: navigationController,
-                               logger: logger)
+        if CoordinatorFlagManager.isLibraryCoordinatorEnabled && !(bookmarkCell is BookmarkItemData) {
+            bookmarkCoordinatorDelegate?.start(from: bookmarkCell)
```

Comment:
I would suggest this, so when we come to enable `isLibraryCoordinatorEnabled`, then we can remove entirely the method `didSelect(profile:...)` from the `BookmarksFolderCell` protocol  
```suggestion
        if CoordinatorFlagManager.isLibraryCoordinatorEnabled {
            if bookmarkCell is BookmarkItemData {
                // call library panel delegate since we have access to it from this `BookmarksPanel`
            } else {
                bookmarkCoordinatorDelegate?.start(from: bookmarkCell)
            }
```